aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/UpdateHandler.java

```diff
@@ -32,39 +32,71 @@
 
         this.logger = logger;
 
-        final ResourceModel model = request.getDesiredResourceState();
+        final ResourceModel desiredModel = request.getDesiredResourceState();
+        final ResourceModel prevModel = request.getPreviousResourceState();
+
```

Comment:
im wondering if we should do some validation upfront to check if the new model has both ECs and upstreams and then just reject the request. This is purely thinking about the rollback path if we start one operation (remove old ECs) and then fail or something, how do we get back to their good state. If we never performed an action, then there is nothing to rollback.

Maybe CFN handles this nicely for us, I don't know. But I figured I'd call this out.

Reply:
I tested: failing stack update before making an update. What happened was
+ stack failed on the first update operation, nothing was changed
+ upon rollback, nothing actually happened. No more calls to the sdk. The framework knew it had nothing to rollback

I also tested: fail after having done something. Results
+ update stack started by removing an upstream
+ put EC failed 
+ rollback started -> upstream was re-added

It seems like the framework keeps track of what was performed before failure and does a rollback by undoing those steps. So we don't need a check like this if there is a rollback with no changes because the framework will handle this.